frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/diskDialog.ts

```diff
@@ -0,0 +1,88 @@
+import { click } from '@console/shared/src/test-utils/utils';
+import { fillInput, selectOptionByText, getSelectedOptionText } from '../utils/utils';
+import * as view from '../../views/dialogs/diskDialog.view';
+import { applyButton, saveButton } from '../../views/kubevirtDetailView.view';
+import { StorageResource, DiskSourceConfig } from '../utils/types';
+import { DISK_SOURCE, INVALID_DISK_SOURCE_CONFIG } from '../utils/consts';
+import { waitForNoLoaders } from '../../views/wizard.view';
+
+export class DiskDialog {
+  sourceMethods = {
+    [DISK_SOURCE.AttachClonedDisk]: async (sourceConfig: DiskSourceConfig) =>
+      DiskDialog.selectSourceAttachClonedDisk(sourceConfig),
+    [DISK_SOURCE.AttachDisk]: async (sourceConfig: DiskSourceConfig) =>
+      DiskDialog.selectSourceAttachDisk(sourceConfig),
+    [DISK_SOURCE.Container]: async (sourceConfig: DiskSourceConfig) =>
+      DiskDialog.selectSourceContainer(sourceConfig),
+    [DISK_SOURCE.Url]: async (sourceConfig: DiskSourceConfig) =>
```

Comment:
can't we reference the functions directly? `[DISK_SOURCE.Url]: DiskDialog.selectSourceURL`

Reply:
Indeed, this looks better